scripts/test-residual.js

```diff
@@ -170,14 +182,81 @@ function run() {
     if (path.basename(test.name)[0] === ".") continue;
     if (test.name.endsWith("~")) continue;
     if (test.file.includes("// skip")) continue;
+    if (!test.name.includes(args.filter)) continue;
 
     total++;
-    if (runTest(test.name, test.file)) passed++;
+    if (runTest(test.name, test.file, args)) passed++;
     else failed++;
   }
 
   console.log("Passed:", `${passed}/${total}`, (Math.floor(passed / total * 100) || 0) + "%");
   return failed === 0;
 }
 
-if (!run()) process.exit(1);
+// Object to store all command line arguments
+class ProgramArgs {
+  verbose: boolean;
+  filter: string;
+  constructor(verbose: boolean, filter: string) {
+    this.verbose = verbose;
```

Comment:
nit: verbose often means debug output, but in this case it seems to mean "dump code".

Reply:
Dumping the code is used primarily for debugging purposes. Also, this in not breaking new ground, but follows the precedent set by test-internal.js.